packages/core/src/lib/upload.ts

```diff
@@ -14,25 +18,29 @@ export type UploadOptions = Pick<CoreConfig, 'upload' | 'persist'>;
 export async function upload(
   options: UploadOptions,
   uploadFn: typeof uploadToPortal = uploadToPortal,
-) {
+): Promise<MultipleFileResults> {
   if (options?.upload === undefined) {
     throw new Error('upload config needs to be set');
   }
 
   const { apiKey, server, organization, project } = options.upload;
-  const { outputDir } = options.persist;
-  const report = reportSchema.parse(
-    JSON.parse(readFileSync(join(outputDir, 'report.json')).toString()),
-  );
+  const reports = loadReports({ ...options.persist, format: ['json'] });
 
-  const data = {
-    organization,
-    project,
-    commit: await latestHash(),
-    ...jsonToGql(report),
-  };
+  const uploadResults: MultipleFileResults = await Promise.allSettled(
+    reports.map(async ([filename, reportContent]: [string, string]) => {
+      const report = reportSchema.parse(JSON.parse(reportContent));
+      const data = {
+        organization,
+        project,
+        commit: await latestHash(),
+        ...jsonToGql(report),
+      };
+      return uploadFn({ apiKey, server, data }).then(
+        () => [filename] satisfies FileResult,
+      );
+    }),
+  );
 
-  return uploadFn({ apiKey, server, data }).catch(e => {
-    throw new Error('upload failed. ' + e.message);
-  });
+  logMultipleFileResults(uploadResults, 'Uploaded reports');
+  return uploadResults;
```

Comment:
OK, I have problems with this:

1. This is out of scope for this PR. What has it got to do with adding `--persist.filename`?
2. Do we even want this? I don't remember ever discussing it. I don't get why `upload` should be searching for report files instead of uploading the single report file specified in the config. To me that's really unexpected behaviour, and you could easily end up re-uploading reports if the output directory isn't cleaned up.

Reply:
I set the default report file name back to 'report.json' and move a time stamp file into another issue.